src/Tasks.UnitTests/WriteCodeFragment_Tests.cs

```diff
@@ -137,6 +137,31 @@ public void FileWithPathAndDirectoryDoesNotExist()
             Assert.True(File.Exists(task.OutputFile.ItemSpec));
         }
 
+        /// <summary>
+        /// File name is set but no OutputDirectory
+        /// </summary>
+        [Fact]
+        public void FileNameNoDirectory()
+        {
+            WriteCodeFragment task = new WriteCodeFragment();
+            MockEngine engine = new MockEngine(true);
+            task.BuildEngine = engine;
+            task.Language = "c#";
+            task.AssemblyAttributes = new TaskItem[] { new TaskItem("aa") };
+
+            string fileName = "file.tmp";
+            string file = Path.GetFullPath(fileName);
+            task.OutputFile = new TaskItem(fileName);
+            bool result = task.Execute();
+
+            Assert.True(result);
+
+            Assert.Equal(file, task.OutputFile.ItemSpec);
+            Assert.True(File.Exists(file));
```

Comment:
Please wrap the test in a TestEnvironment, and use `ExpectFile` to get the path of the file and assert its existence--that way cleanup can happen in "the usual way".